### PR TITLE
Add DetectionVOCEvaluator

### PR DESCRIPTION
**Pull request by yuyu2172**
The tests for the new class follows the Chainer Evaluator's tests.
https://github.com/pfnet/chainer/blob/_v2/tests/chainer_tests/training_tests/extensions_tests/test_evaluator.py

### CODE REVIEW

File: chainercv/extensions/detection/detection_voc_evaluator.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            following tuple :obj:`img, bbox, label` or
+            :obj:`img, bbox, label, difficult`.
+            :obj:`img` is an image, :obj:`bbox` is coordinates of bounding
+            boxes, :obj:`label` is labels of the bounding boxes and
+            :obj:`difficult` is whether the bounding boxes are difficult or
+            not. If :obj:`difficult` is returned, difficult ground truth
+            will be ignored from evaluation.
+        target (chainer.Link): An detection link. This link must have
+            :meth:`predict` method which take a list of images and returns
+            :obj:`bboxes`, :obj:`labels` and :obj:`scores`.
```
Comment (Hakuyume) [inline chainercv/extensions/detection/detection_voc_evaluator.py]: Please add `use_07_metric` to `Args` section.

File: tests/extensions_tests/detection_tests/test_detection_voc_evaluator.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+import chainer
+from chainer.datasets import TupleDataset
+from chainer.iterators import SerialIterator
+from chainer import testing
+
+from chainercv.extensions import DetectionVOCEvaluator
+
+
+def _generate_bbox(xp, n, img_size, min_length, max_length):
```
Comment (Hakuyume) [inline tests/extensions_tests/detection_tests/test_detection_voc_evaluator.py]: `xp` is always `np` in this test.

File: chainercv/extensions/detection/detection_voc_evaluator.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        iterator (chainer.Iterator): An iterator. Each sample should be
+            following tuple :obj:`img, bbox, label` or
+            :obj:`img, bbox, label, difficult`.
+            :obj:`img` is an image, :obj:`bbox` is coordinates of bounding
+            boxes, :obj:`label` is labels of the bounding boxes and
+            :obj:`difficult` is whether the bounding boxes are difficult or
+            not. If :obj:`difficult` is returned, difficult ground truth
+            will be ignored from evaluation.
+        target (chainer.Link): An detection link. This link must have
+            :meth:`predict` method which take a list of images and returns
```
Comment (Hakuyume) [inline chainercv/extensions/detection/detection_voc_evaluator.py]: `which take` -> `which takes`

File: chainercv/extensions/detection/detection_voc_evaluator.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+from chainercv.evaluations import eval_detection_voc
+from chainercv.utils import apply_detection_link
+
+
+class DetectionVOCEvaluator(chainer.training.extensions.Evaluator):
+
+    """An extension that evaluates a detection model by PASCAL VOC metric.
+
+    This extension iterates over an iterator and evaluate the prediction
```
Comment (Hakuyume) [inline chainercv/extensions/detection/detection_voc_evaluator.py]: `evaluate` -> `evaluates`

File: chainercv/extensions/__init__.py [hunk 5]
```diff
@@ -1 +1,2 @@
 from chainercv.extensions.detection.detection_vis_report import DetectionVisReport  # NOQA
+from chainercv.extensions.detection.detection_voc_evaluator import DetectionVOCEvaluator  # NOQA
```
Comment (Hakuyume) [inline chainercv/extensions/__init__.py]: How about `DetectionEvaluatorVOC`? When we add other metrics, we can write `eval_detection_*` and `DetectionEvaluator*`.

Comment (yuyu2172) [inline chainercv/extensions/__init__.py]: Nice suggestion.

Comment (yuyu2172) [inline chainercv/extensions/__init__.py]: But, I think that `*<TYPE_NAME>` is a ubiquitous rule. (e.g. TransformDataset is a dataset class)
Do you think it is worth it to break this rule?

Comment (Hakuyume) [inline chainercv/extensions/__init__.py]: I see. `DetectionVOCEvaluator` is OK.

**Comment by Hakuyume** [comment]
LGTM

**Pull request merged**